internal/controller/scheduler/bk_job_checker.go

```diff
@@ -0,0 +1,165 @@
+package scheduler
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/buildkite/agent-stack-k8s/v2/api"
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// BkJobChecker monitors Buildkite jobs for cancellation state changes.
+type BkJobChecker struct {
```

Comment:
Hmm. I think perhaps one of these. "Buildkite" tends to be shortened to "BK", not "Bk", don't you think?

```suggestion
type BKJobChecker struct {
```

```suggestion
type BuildkiteJobChecker struct {
```